Bump elixir/erlang and dependencies & add timezone to airports

dgigafox opened the pull request:
Changes:
- Bump elixir from 1.16 to 1.17 and erlang from 26.2 to 27.0
- Add timezone data accessible through the `timezone_id` field of the Airport struct which can be updated using `mix airports.update`. The source of timezone data is [tz_world](https://github.com/kimlai/tz_world)
- Bump other mix dependencies

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump dialyxir from 1.1.0 to 1.2.0

Bumps [dialyxir](https://github.com/jeremyjh/dialyxir) from 1.1.0 to 1.2.0.
- [Release notes](https://github.com/jeremyjh/dialyxir/releases)
- [Changelog](https://github.com/jeremyjh/dialyxir/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jeremyjh/dialyxir/compare/1.1.0...1.2.0)

---
updated-dependencies:
- dependency-name: dialyxir
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

samhamilton committed:
Merge pull request #4 from martide/dependabot/github_actions/actions/checkout-3

Bump actions/checkout from 2 to 3

samhamilton committed:
Merge pull request #8 from martide/dependabot/hex/dialyxir-1.2.0

Bump dialyxir from 1.1.0 to 1.2.0

dependabot[bot] committed:
Bump ex_doc from 0.28.0 to 0.28.5 (#10)

dependabot[bot] committed:
Bump excoveralls from 0.14.4 to 0.14.6 (#9)

dependabot[bot] committed:
Bump excoveralls from 0.14.6 to 0.16.1

Bumps [excoveralls](https://github.com/parroty/excoveralls) from 0.14.6 to 0.16.1.
- [Release notes](https://github.com/parroty/excoveralls/releases)
- [Changelog](https://github.com/parroty/excoveralls/blob/master/CHANGELOG.md)
- [Commits](https://github.com/parroty/excoveralls/compare/v0.14.6...v0.16.1)

---
updated-dependencies:
- dependency-name: excoveralls
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

samhamilton committed:
Merge pull request #17 from martide/dependabot/hex/excoveralls-0.16.1

Bump excoveralls from 0.14.6 to 0.16.1

dgigafox committed:
Skip header when parsing csv

samhamilton committed:
Merge pull request #19 from martide/fix/remove-header

Skip header when parsing CSV

dependabot[bot] committed:
Bump dialyxir from 1.2.0 to 1.3.0 (#18)

dependabot[bot] committed:
Bump ex_doc from 0.28.5 to 0.29.4 (#16)

dependabot[bot] committed:
Bump ex_doc from 0.29.4 to 0.30.6

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.29.4 to 0.30.6.
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/compare/v0.29.4...v0.30.6)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

samhamilton committed:
Merge pull request #24 from martide/dependabot/github_actions/actions/checkout-4

Bump actions/checkout from 3 to 4

samhamilton committed:
Merge pull request #22 from martide/dependabot/hex/ex_doc-0.30.6

Bump ex_doc from 0.29.4 to 0.30.6

samhamilton committed:
Bump deps

samhamilton committed:
add castore

dgigafox committed:
Update source csv

dgigafox committed:
Update tool versions

samhamilton committed:
Merge pull request #28 from martide/feat/update-csv-source

feat(WEB-0): Update csv source

dependabot[bot] committed:
Bump ex_doc from 0.31.0 to 0.31.1

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.31.0 to 0.31.1.
- [Release notes](https://github.com/elixir-lang/ex_doc/releases)
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/compare/v0.31.0...v0.31.1)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

samhamilton committed:
Merge pull request #29 from martide/dependabot/hex/ex_doc-0.31.1

Bump ex_doc from 0.31.0 to 0.31.1

dependabot[bot] committed:
Bump ex_doc from 0.31.1 to 0.31.2 (#30)

dependabot[bot] committed:
Bump castore from 1.0.5 to 1.0.6 (#31)

samhamilton committed:
update tests

samhamilton committed:
Merge pull request #34 from martide/gh

update tests

samhamilton committed:
Bump packages

dgigafox committed:
Add API to get timezone based on airport code

dgigafox committed:
Save timezone to source file for faster retrieve

alexfilippovw committed:
Merge pull request #37 from martide/feat/timezones

feat(WEB-7334): Add API for getting timezone based on IATA code

dependabot[bot] committed:
Bump ex_doc from 0.32.0 to 0.34.1 (#39)

dependabot[bot] committed:
Bump castore from 1.0.6 to 1.0.7 (#36)

samhamilton committed:
Merge branch 'nerds-and-company:master' into master

dependabot[bot] committed:
Bump ex_doc from 0.34.1 to 0.34.2

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.34.1 to 0.34.2.
- [Release notes](https://github.com/elixir-lang/ex_doc/releases)
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/compare/v0.34.1...v0.34.2)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump castore from 1.0.7 to 1.0.8

Bumps [castore](https://github.com/elixir-mint/castore) from 1.0.7 to 1.0.8.
- [Commits](https://github.com/elixir-mint/castore/compare/v1.0.7...v1.0.8)

---
updated-dependencies:
- dependency-name: castore
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dgigafox committed:
Bump elixir to 1.17 and erlang to 27.0

dgigafox committed:
Fix github workflow

samhamilton committed:
Merge pull request #42 from martide/bump-elixir-erlang

Bump elixir from 1.16 to 1.17 and erlang from 26.2 to 27.0

dgigafox committed:
Merge pull request #41 from martide/dependabot/hex/castore-1.0.8

Bump castore from 1.0.7 to 1.0.8

dgigafox committed:
Merge pull request #40 from martide/dependabot/hex/ex_doc-0.34.2

Bump ex_doc from 0.34.1 to 0.34.2

sjoulbak commented on mix.exs:
```diff
@@ -27,10 +28,12 @@ defmodule Airports.Mixfile do
   defp deps do
     [
-      {:nimble_csv, "~> 1.1"},
+      {:castore, "~> 1.0"},
```
I don't see this package being used. Can it be removed?

dgigafox replied:
Yup, I think we can remove it. Will push here

dgigafox replied:
Seems we can't remove it as coveralls are using it for OTP 24 and below:
```
** (ExCoveralls.ReportUploadError) Failed to use any SSL certificates. We didn't find the :castore library,
and we couldn't use OS certificates because that requires OTP 25 or later.
If you want to use :castore, please add

  {:castore, "~> 1.0"}
```

sjoulbak replied:
Ah check! Can you change it to `{:castore, "~> 1.0", only: :test},`? `excoveralls` seems only to be used at testing.

sjoulbak commented on README.md:
````diff
@@ -13,6 +13,14 @@ def deps do
+Airports depends on [tz_world](https://github.com/kimlai/tz_world) for timezone data. To install airports with timezone data, run:
+
+```elixir
+mix airports.update
````
I like the addition of timezones! For confirmation, is it only necessary to run this command after `priv/airports.csv` has been changed? Or is this something users of the package have to do at their Elixir project as well?

dgigafox replied:
Only when `priv/airports.csv` changed since the `airports_with_tz.csv` file is already included in the package

sjoulbak commented on lib/airport.ex:
```diff
@@ -16,6 +17,7 @@ defmodule Airports.Airport do
     :wikipedia_link,
-    :keywords
+    :keywords,
+    :timezone_id
```
Could you add a changelog entry at 1.0.0 for this addition?

dgigafox replied:
Sure thing

sjoulbak commented on lib/airports/loader.ex:
```diff
@@ -0,0 +1,63 @@
+  def run do
+    source_path()
+    |> File.stream!()
+    |> AirportsParser.parse_stream(skip_headers: true)
```
Could you add a changelog entry for this addition as well?

dgigafox replied:
Sure thing

sjoulbak reviewed (commented):
Hey @dgigafox 👋 Thanks for your contribution! I ran `mix airports.update` locally and liked the progress indicator. I did not had any changes, so its looking good 👌 A nice addition for in the future would be to first update `priv/airports.csv` in the same command 😄 

I left a few comments, could you take a look at those?

sjoulbak committed:
chore(github actions): let elixir workflow run on push and pull_request

sjoulbak commented on .github/workflows/elixir.yml:
Can you make this branch up-to-date with its base branch? I added `pull_request` to the github action workflow, so it will run here as well.

dgigafox replied:
Sure thing

dgigafox replied:
Having a problem with github action on push and pull request as it doubles the job and I'm getting this error:
```
** (ExCoveralls.ReportUploadError) Failed to upload the report to 'https://coveralls.io/' (reason: status_code = 422, body = {"message":"Can't add a job to a build that is already closed. Build 4a906ff5af85935d69cae47d180084dacce7e615-PR-43 is closed. See docs.coveralls.io/parallel-builds","error":true}).
```
![image](https://github.com/user-attachments/assets/3151dbae-4e35-493d-ab57-fae820ff8288)

@samhamilton could you take a look at this one too?

sjoulbak replied:
I think this is caused by the following change: https://github.com/nerds-and-company/airports/commit/d23b5b9ca5e36ef8c3bc8bbb7504f7af4bd08b82

sjoulbak replied:
@dgigafox I pushed a new change to the branch. Hopefully that will fix it. Can you make this branch up-to-date with its base branch again?

sjoulbak committed:
fix(github actions): prevent actions being ran twice

Inspired by: https://github.com/orgs/community/discussions/26276#discussioncomment-3251154

dgigafox committed:
Update CHANGELOG

dgigafox committed:
Run castore on test only

dgigafox committed:
Merge pull request #43 from martide/change-request-from-upstream

Change requests from upstream

dgigafox commented:
@sjoulbak change requests are pushed. Please take a look again. Thanks!

Btw, about
> A nice addition for in the future would be to first update priv/airports.csv in the same command 😄

Agree on this. Will take a look at that

dgigafox commented:
@sjoulbak error on workflow appeared again because I think this is both a pull request and a pull to master.

sjoulbak commented:
@dgigafox Its fine by me if you remove excoveralls from the workflow for now, but keep it in the code base so we can run it locally. Currently we're not covering a lot for this package

dgigafox committed:
Replace coveralls with test

samhamilton committed:
Merge pull request #44 from martide/remove-coveralls-from-workflow

Replace coveralls with test

dgigafox commented:
> @dgigafox Its fine by me if you remove excoveralls from the workflow for now, but keep it in the code base so we can run it locally. Currently we're not covering a lot for this package

@sjoulbak We replaced the coveralls with a simple `mix test`.  All green now

sjoulbak reviewed (approved):
🎉

sjoulbak commented:
@dgigafox and @samhamilton Thanks both for all of your effort! 🎉 I've released 1.0.0 🎉
